nova/templates/hook-scripts/preStop-nova-compute.py

```diff
@@ -0,0 +1,129 @@
+#!/bin/python
+import kubernetes.config
+import os
+import socket
+import time
+import yaml
+
+from kubernetes import client as k8sclient
+from keystoneauth1 import identity, session
+from novaclient import client, exceptions
+
+OPENRC_PATH = "/test.yaml"
+
+
+def load_configs(openrc_parh=OPENRC_PATH):
+    with open(openrc_parh, 'r') as stream:
+            try:
+                config = yaml.load(stream)
+            except yaml.YAMLError as e:
+                print "Cannot load configs: {}".format(e)
+                os._exit(1)
+
+    return config
+
+
+def get_sessions():
+    config = load_configs()
+    try:
+        auth = identity.v3.Password(
+            auth_url=config["OS_AUTH_URL"],
+            username=config["OS_USERNAME"],
+            password=config["OS_PASSWORD"],
+            project_name=config["OS_PROJECT_NAME"],
+            user_domain_id=config["OS_USER_DOMAIN_ID"],
+            project_domain_id=config["OS_PROJECT_DOMAIN_ID"]
+        )
+        keystone_session = session.Session(auth=auth)
+    except Exception as e:
+        print "Cannot create session: {}".format(e)
+        os._exit(1)
+
+    return keystone_session
+
+
+def get_nova_client():
+    keystone_session = get_sessions()
+    try:
+        nova_client = client.Client(
+            version="2.1",
+            session=keystone_session
```

Comment:
Please add `endpoint_type="internalURL"` otherwise it won't work when Ingress is enabled.